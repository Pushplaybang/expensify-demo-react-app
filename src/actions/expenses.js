import uuid from "uuid";
import moment from "moment";
// ADD_EXPENSE
export const addExpense = ({
  description = "",
  note = "",
  amount = 0,
  createdAt = moment().valueOf()
} = {}) => ({
  type: "ADD_EXPENSE",
  expense: {
    id: uuid(),
    description,
    note,
    amount,
    createdAt
  }
});
// EDIT_EXPENSE
export const editExpense = (id, updates) => ({
  type: "EDIT_EXPENSE",
  id,
  updates
});
// REMOVE_EXPENSE
export const removeExpense = ({ id } = {}) => ({
  type: "REMOVE_EXPENSE",
  id
});
