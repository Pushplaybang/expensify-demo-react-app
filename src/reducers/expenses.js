export const expensesReducer = (state = [], action) => {
  switch (action.type) {
    case "ADD_EXPENSE":
      return [...state, action.expense];
    case "EDIT_EXPENSE":
      return state.map(expense => {
        if (expense.id !== action.id) return expense;
        return { ...expense, ...action.updates };
      });
    case "REMOVE_EXPENSE":
      return state.filter(expense => expense.id !== action.id);
    case "SET_EXPENSES":
      return action.expenses;
    default:
      return state;
  }
};
