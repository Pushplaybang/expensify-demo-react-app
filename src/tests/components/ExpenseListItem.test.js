import React from "react";
import { shallow } from "enzyme";
import expenses from "./../fixtures/expenses";
import ExpenseListItem from "./../../components/ExpenseListItem";

test("Should render the expense list item with expense data", () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
  expect(wrapper).toMatchSnapshot();
});
 