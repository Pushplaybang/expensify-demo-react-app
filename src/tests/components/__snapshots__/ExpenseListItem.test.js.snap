// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the expense list item with expense data 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      item 1 aabb
    </h3>
  </Link>
  <p>
    100
     - 
    0
  </p>
</div>
`;
